schema/RequestSignInLinkMutation.re

```diff
@@ -18,7 +20,39 @@ let requestSignInLink =
       "requestSignInLink",
       ~typ=non_null(GraphqlTypes.errorPayloadType),
       ~args=Arg.[arg("input", ~typ=non_null(requestSignInLinkInput))],
-      ~resolve=(_, (), input) =>
-      Lwt.return(Ok(None))
+      ~resolve=(context: Context.t, (), email) =>
+      switch (User.getByEmail(context.connection, email)) {
+      | Some(user) =>
+        Email.sendEmail(
+          ~from="contact@summhub.com",
+          ~to_=email,
+          ~subject="Summhub - Login Access Link",
+          ~content=
+            "Your access link is summhub://access-link/"
+            ++ Auth.encodeToken(user.id),
+        )
+        |> ignore;
+        Lwt.return(Ok(None));
+      | None =>
+        let result = User.createUser(context.connection, email);
+        switch (result) {
+        | None => Lwt.return(Ok(Some(Errors.somethingWentWrong)))
+        | Some(_) =>
+          switch (User.getByEmail(context.connection, email)) {
+          | Some(user) =>
+            Email.sendEmail(
```

Comment:
can be a helper on User model for reuse

Reply:
What? the send auth email?

Reply:
yes, because we can reuse in the future for other authing. I mean this because this is part of the "account" functionality